lib/SimpleSAML/Utils/Config/Metadata.php

```diff
@@ -108,6 +115,13 @@ function ($t) {
             throw new \InvalidArgumentException('"contactType" is mandatory and must be one of '.$types.".");
         }
 
+        // check attributes is an associative array
+        if (isset($contact['attributes'])) {
+            if (empty($contact['attributes']) || !is_array($contact['attributes']) || count(array_filter(array_keys($contact['attributes']), 'is_string')) == 0) {
```

Comment:
As a rule of thumb, strict comparisons (`===`) are always preferred to keep us in the safe side in case of an unexpected return value. In this case it should be ok, but the strict comparison could protect us from future changes in the signature of `count()`.

Also, I suspect the line is a bit too long. Can you split it to make it a bit more readable? I.e.:

```php
if (empty($contact['attributes']) || !is_array($contact['attributes']) || 
    count(array_filter(array_keys($contact['attributes']), 'is_string')) == 0
) {
```